### ISSUE TITLE
Need apply_angular_impulse

### ISSUE THREAD

**Issue by buckle2000** [godotengine/godot#4475]
I think a `RigidBody` should have both **mass** and **moment**.
Also, they need `apply_angular_impulse` to change angular velocity based on their moment.

**Comment by akien-mga** [comment]
Please be more specific.

**Comment by buckle2000** [comment]
@akien-mga updated

**Comment by akien-mga** [comment]
There are many objects in Godot, it would have been nice to mention what objects you are talking about. I can only assume you're talking about RigidBody{,2D}.

They already have a mass and moment of inertia. The latter is computed based on the mass and the shape attached to the body; PR #4387 adds the possibility to set an arbitrary moment of inertia (i.e. not depending on the mass and shape). Not sure about ``apply_angular_impulse``, maybe @JoshuaGrams would have a more educated opinion.

**Comment by slapin** [comment]
Shape dependency is stupid thing - it is used for collision, so sets volume, also using it for momentum
calculation causes weird effects, especially with car physics.
If you want to depend on shape, that is fine as long as it is not used for collision.

**Comment by slapin** [comment]
Angular impulse can be an nice addition as we have angular velocity, for "full stack"
But I think such basic things should be implemented in root physocs objects, not only RigidBody.
Or VehicleBody and some others should inherit from RigidBody (or RigidBody shold have vehicle mode which would solve all problems).

**Comment by slapin** [comment]
To be clear - the shapes when attached to physics body solve 2 completely different problems - 
a) collisions - to prevent collisions with objects
b) momentum calculations.
These 2 are too different uses. I.e. when you need to draw a good collision around your car
as you do you usually see that it really doesn't work as car physics explodes. So you have to draw bigger collision, which is insane, and will make your car behave unrealistically in case of collisions.
So we either need a (stupid) multiplier, (shitty) conversion formula, or just use different shapes for collisions, allowing people to play with shapes as they wish without affecting physics too much.

**Comment by buckle2000** [comment]
@slapin exactly, I need angular impulse to make a vehicle-like thing.
Also, seems like this proposal is already [here](https://github.com/godotengine/godot/issues/1058#issuecomment-68261961). Closing.

**Comment by JoshuaGrams** [comment]
@buckle2000 actually, that comment on #1058 is wrong. It's a really common mistake people make when they're first looking at physics: we've all made it. It's somewhat counter-intuitive because it's a difficult situation to replicate in real-life (there's almost always friction and other forces involved).

Let's say you had a puck floating on an air-hockey table, and a little focused air-jet gun which could apply a constant force. If you aim it at the center of the puck, it will accelerate forward at `force/mass`. If you aim it at the edge of the puck, it will _still_ accelerate at `force/mass`, but because you're on the edge it will _also_ start spinning. So the edge that you're pushing will move farther, and thus more work is done (work is force times distance).

TL;DR version: you **do** need angular impulse. And it's fairly easy to add. You need the 3D version, right?

**Comment by Calinou** [comment]
Is this covered by the RigidBody `add_torque()` method?

**Comment by JoshuaGrams** [comment]
@Calinou it looks like there's also [RigidBody.apply_torque_impulse](https://docs.godotengine.org/en/latest/classes/class_rigidbody.html#class-rigidbody-method-apply-torque-impulse), so yeah, this can be closed. Assuming both of them actually work. And hey, it looks like RigidBody2D has both as well. What a concept!

**Comment by Calinou** [comment]
Alright, closing then :slightly_smiling_face:
